sidechain/consensus/aura/src/mock.rs

```diff
@@ -58,10 +69,10 @@ impl Environment<ParentchainBlock, SignedSidechainBlock> for EnvironmentMock {
 
 	fn init(
 		&mut self,
-		_: Header,
+		header: Header,
 		_: ShardIdentifierFor<SignedSidechainBlock>,
 	) -> Result<Self::Proposer, Self::Error> {
-		Ok(ProposerMock)
+		Ok(ProposerMock { parentchain_header: header })
```

Comment:
The proposer mock now stores the parentchain header that was passed in, and proposes a sidechain block based on this parentchain header.